lib/irb/debug.rb

```diff
@@ -57,6 +57,22 @@ def DEBUGGER__.capture_frames(*args)
           DEBUGGER__::ThreadClient.prepend(SkipPathHelperForIRB)
         end
 
+        unless DEBUGGER__::CONFIG[:no_hint]
+          irb_output_modifier_proc = Reline.output_modifier_proc
```

Comment:
This part is called multiple times (when `continue` hits next `binding.irb`)
```ruby
# a.rb
5.times do
  binding.irb
end
```

```
ruby -I path/to/irb/lib a.rb
irb(main):001> debug
(called)
irb:rdbg(main):002> c # debug command
(called)
irb:rdbg(main):003> c # debug command # debug command
(called)
irb:rdbg(main):004> c # debug command # debug command # debug command
```
And in this case, irb_output_modifier_proc is chained. irb_output_modifier_proc calls previous irb_output_modifier_proc.

I think moving irb's original output_modifier_proc to a method and call it from here is be better.

Reply:
I understand that'd be the case, but I actually can't reproduce it 🤔 

```
irb(main):001> debug
irb:rdbg(main):002> c # debug command

From: test.rb @ line 2 :

    1: 5.times do
 => 2:   binding.irb
    3: end

irb:rdbg(main):003> c # debug command

From: test.rb @ line 2 :

    1: 5.times do
 => 2:   binding.irb
    3: end

irb:rdbg(main):004> c # debug command

From: test.rb @ line 2 :

    1: 5.times do
 => 2:   binding.irb
    3: end

irb:rdbg(main):005> c # debug command

From: test.rb @ line 2 :

    1: 5.times do
 => 2:   binding.irb
    3: end

irb:rdbg(main):006>
```

> I think moving irb's original output_modifier_proc to a method and call it from here is better.

Will do 👍

Reply:
I forgot to write, it can be reproduced after
```
output = output.sub(/\n$/, " # debug command")
```
is changed to 
```
output = output.sub(/\n$/, " # debug command\n")
```

Reply:
I ended up using a flag to check if the `debug` modifier block has been defined or not instead. Calling an input method object's method here feels like a bad coupling.